core/trino-main/src/main/java/io/trino/sql/planner/iterative/rule/ApplyTableScanRedirection.java

```diff
@@ -121,7 +121,7 @@ public Result apply(TableScanNode scanNode, Captures captures, Context context)
                                 destinationTable,
```

Comment:
first 2 commits can be separate PR, thus they would land faster.

Reply:
Created PR https://github.com/trinodb/trino/pull/7365
Note that the 1st commit is a pre-requisite for the 3rd commit here.